File: software/examples/delphi/ExampleThreshold.pas
program ExampleThreshold;

{$ifdef MSWINDOWS}{$apptype CONSOLE}{$endif}
{$ifdef FPC}{$mode OBJFPC}{$H+}{$endif}

uses
  SysUtils, IPConnection, BrickletLine;

type
  TExample = class
  private
    ipcon: TIPConnection;
    line: TBrickletLine;
  public
    procedure ReachedCB(sender: TBrickletLine; const reflectivity: word);
    procedure Execute;
  end;

const
  HOST = 'localhost';
  PORT = 4223;
  UID = 'XYZ'; { Change to your UID }

var
  e: TExample;

{ Callback for reflectivity greater than 2000 }
procedure TExample.ReachedCB(sender: TBrickletLine; const reflectivity: word);
begin
  WriteLn(Format('Reflectivity: %d', [reflectivity]));
end;

procedure TExample.Execute;
begin
  { Create IP connection }
  ipcon := TIPConnection.Create;

  { Create device object }
  line := TBrickletLine.Create(UID, ipcon);

  { Connect to brickd }
  ipcon.Connect(HOST, PORT);
  { Don't use device before ipcon is connected }

  { Get threshold callbacks with a debounce time of 1 seconds (1000ms) }
  line.SetDebouncePeriod(1000);

  { Register threshold reached callback to procedure ReachedCB }
  line.OnReflectivityReached := {$ifdef FPC}@{$endif}ReachedCB;

  { Configure threshold for "greater than 2000" }
  line.SetReflectivityCallbackThreshold('>', 2000, 0);

  WriteLn('Press key to exit');
  ReadLn;
  ipcon.Destroy; { Calls ipcon.Disconnect internally }
end;

begin
  e := TExample.Create;
  e.Execute;
  e.Destroy;
end.

File: software/examples/vbnet/ExampleThreshold.vb
Imports System
Imports Tinkerforge

Module ExampleThreshold
    Const HOST As String = "localhost"
    Const PORT As Integer = 4223
    Const UID As String = "XYZ" ' Change XYZ to the UID of your Line Bricklet

    ' Callback subroutine for reflectivity reached callback
    Sub ReflectivityReachedCB(ByVal sender As BrickletLine, _
                              ByVal reflectivity As Integer)
        Console.WriteLine("Reflectivity: " + reflectivity.ToString())
    End Sub

    Sub Main()
        Dim ipcon As New IPConnection() ' Create IP connection
        Dim l As New BrickletLine(UID, ipcon) ' Create device object

        ipcon.Connect(HOST, PORT) ' Connect to brickd
        ' Don't use device before ipcon is connected

        ' Get threshold callbacks with a debounce time of 1 second (1000ms)
        l.SetDebouncePeriod(1000)

        ' Register reflectivity reached callback to subroutine ReflectivityReachedCB
        AddHandler l.ReflectivityReached, AddressOf ReflectivityReachedCB

        ' Configure threshold for reflectivity "greater than 2000"
        l.SetReflectivityCallbackThreshold(">"C, 2000, 0)

        Console.WriteLine("Press key to exit")
        Console.ReadLine()
        ipcon.Disconnect()
    End Sub
End Module
